Sources/TuistSigning/Certificate/CertificateParser.swift

```diff
@@ -0,0 +1,36 @@
+import Foundation
+import TuistSupport
+import TSCBasic
+
+protocol CertificateParsing {
+    func parse(publicKey: AbsolutePath, privateKey: AbsolutePath) throws -> Certificate
+}
+
+final class CertificateParser: CertificateParsing {
+    func parse(publicKey: AbsolutePath, privateKey: AbsolutePath) throws -> Certificate {
+        let subject = try self.subject(at: publicKey)
+        let isRevoked = subject.contains("REVOKED")
+        
+        let nameRegex = try NSRegularExpression(pattern: "CN=([^/]+)/", options: [])
+        guard let result = nameRegex.firstMatch(in: subject, options: [], range: NSRange(location: 0, length: subject.count)) else { fatalError() }
+        let name = NSString(string: subject).substring(with: result.range(at: 1)).spm_chomp()
+        
+        let developmentTeamRegex = try NSRegularExpression(pattern: "OU=([^/]+)/", options: [])
+        guard
+            let developmentTeamResult = developmentTeamRegex.firstMatch(in: subject, options: [], range: NSRange(location: 0, length: subject.count))
+        else { fatalError() }
```

Comment:
A fatal error here is a terrible experience for the developer. I'd rather raise an error saying something along the lines of:
```
We couldn't parse the development team while parsing the following output from the file xxx: yyy
```

Reply:
Yeah, I know, there are still a lot rough edges, I just wanted to validate that the basic implementation is going in the right direction 👍